Input/InputFieldSpotlightDecorator.js

```diff
@@ -141,78 +141,84 @@ const InputSpotlightDecorator = hoc(defaultConfig, (config, Wrapped) => {
 		constructor (props) {
 			super(props);
 
-			this.state = {
-				focused: null,
-				node: null,
-				fromMouse: false
-			};
-
+			this.focused = null;
+			this.node = null;
+			this.fromMouse = false;
+			this.prevFocused = null;
+			this.prevNode = null;
 			this.ariaHidden = props.noReadoutOnFocus || null;
 			this.paused = new Pause('InputSpotlightDecorator');
 			this.handleKeyDown = handleKeyDown.bind(this);
 		}
 
-		componentDidUpdate (_, prevState) {
-			this.ariaHidden = null;
-			this.updateFocus(prevState);
-		}
-
 		componentWillUnmount () {
 			this.paused.resume();
 
-			if (this.state.focused === 'input') {
+			if (this.focused === 'input') {
 				const {onSpotlightDisappear} = this.props;
 
 				if (onSpotlightDisappear) {
 					onSpotlightDisappear();
 				}
 
 				if (!noLockPointer) {
-					releasePointer(this.state.node);
+					releasePointer(this.node);
 				}
 			}
 		}
 
-		updateFocus = (prevState) => {
+		updateComponent () {
+			this.ariaHidden = null;
+			this.updateFocus(this.prevFocused, this.prevNode);
+			this.prevFocused = this.focused;
+			this.prevNode = this.node;
+		}
+
+		updateFocus = (prevFocused, prevNode) => {
 			// focus node if `InputSpotlightDecorator` is pausing Spotlight or if Spotlight is paused
 			if (
-				this.state.node &&
-				Spotlight.getCurrent() !== this.state.node &&
+				this.node &&
+				Spotlight.getCurrent() !== this.node &&
 				(this.paused.isPaused() || !Spotlight.isPaused())
 			) {
-				if (this.state.fromMouse) {
-					this.state.node.focus({preventScroll: true});
+				if (this.fromMouse) {
+					this.node.focus({preventScroll: true});
 				} else {
-					this.state.node.focus();
+					this.node.focus();
 				}
 			}
 
-			const focusChanged = this.state.focused !== prevState.focused;
+			const focusChanged = this.focused !== prevFocused;
 			if (focusChanged) {
-				if (this.state.focused === 'input') {
+				if (this.focused === 'input') {
 					forwardCustom('onActivate')(null, this.props);
 					if (!noLockPointer) {
-						lockPointer(this.state.node);
+						lockPointer(this.node);
 					}
 					this.paused.pause();
-				} else if (prevState.focused === 'input') {
+				} else if (prevFocused === 'input') {
 					forwardCustom('onDeactivate')(null, this.props);
 					if (!noLockPointer) {
-						releasePointer(prevState.node);
+						releasePointer(prevNode);
 					}
 					this.paused.resume();
 				}
 			}
 		};
 
 		focus = (focused, node, fromMouse) => {
-			this.setState({focused, node, fromMouse});
+			this.focused = focused;
+			this.node = node;
+			this.fromMouse = fromMouse;
+			this.updateComponent();
 		};
 
 		blur = () => {
-			this.setState((state) => (
-				state.focused || state.node ? {focused: null, node: null} : null
-			));
+			if (this.focused || this.node) {
+				this.focused = null;
+				this.node = null;
+				this.updateComponent();
+			}
```

Comment:
else is missed.

Reply:
Since the effect of setState(null) is not to call render , the else case is unnecessary.